src/seedsigner/gui/components.py

```diff
@@ -568,6 +570,52 @@ def render(self):
             self.icon.render()
 
 
+@dataclass
+class ToastOverlay(BaseComponent):
+    icon_name: str = None
+    color: str = None
+    label_text: str = None
+
+    def __post_init__(self):
+        super().__post_init__()
+            
+        self.icon = Icon(
+            image_draw=self.image_draw,
+            canvas=self.canvas,
+            screen_x=20,
+            screen_y=190,
+            icon_name=self.icon_name,
+            icon_size=30,
+            icon_color=self.color
+        )
+        
+        self.label = TextArea(
+            image_draw=self.image_draw,
+            canvas=self.canvas,
+            text=self.label_text,
+            font_size=19,
+            font_color=self.color,
+            edge_padding=0,
+            is_text_centered=False,
+            auto_line_break=False,
+            width=160,
+            height=20,
+            screen_x=55,
+            screen_y=195,
+            allow_text_overflow=False
+        )
+            
+    def render(self):
```

Comment:
Better to implement this as a `BaseScreen` and override `_render` here. That way when the `ToastOverlay` is invoked, you can just call `display()` and it will paint the screen accordingly.

Just don't `super` the `BaseScreen._render()` since that clears the screen.

For max convention-following: can append `icon` and `label` into `self.components` and ensure that they get rendered using the same `components` iteration found in the `BaseScreen._render()`.